### PR TITLE
Use MKL_jll 2023.2.0

### PR DESCRIPTION
**Pull request by amontoison**
I updated `MKL_jll` today. Let's see if the tests pass.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 MKL_jll = "856f044c-d86e-5d09-b602-aeab76dc8ba7"
 
 [compat]
-MKL_jll = "2022.1"
+MKL_jll = "2023.1"
```
Comment (amontoison) [inline Project.toml:12]:
```suggestion
MKL_jll = "2022.1, 2023"
```

**Comment by ViralBShah** [comment]
Is it worth trying 2023.2?

**Comment by amontoison** [comment]
I opened a PR to generate MKL_jll v2023.2.0.

**Comment by ViralBShah** [comment]
On the mac, these just look like GSVD is getting slightly different answers.

On windows, there's something about a testset mutating ENV and not resettign it. I've never seen that before. https://github.com/JuliaLinearAlgebra/MKL.jl/actions/runs/5786517472/job/15681397696?pr=123#step:6:1139

**Comment by amontoison** [comment]
The good news is that we have different errors with this version of MKL_jll!

**Comment by ViralBShah** [comment]
I don't see threading related issues

**Comment by ViralBShah** [comment]
Do you have a Windows machine to test this on?

**Comment by amontoison** [comment]
Sorry Viral, I don't have a Windows machine.

**Comment by amontoison** [comment]
I propose to open an issue about the failing tests.
I'm currently in Japan for ICIAM 2023 but I can have a look when I will back.

**Comment by ViralBShah** [comment]
I am unable to get MKL to pick up MKL_jll 2023.2. If you force the install of MKL_jll 2023.2, Pkg downloads MKL.jl to 0.3. I don't have any other packages installed.

**Comment by amontoison** [comment]
@ViralBShah
You need a release of MKL.jl that supports MKL_jll.jl 2023.2.0.
Julia downloads MKL.jl v0.3 because it's a version where we don't have a compat bound on MKL_jll.

**Comment by ViralBShah** [comment]
I literally just realized that I had forgotten to do that: https://github.com/JuliaRegistries/General/pull/89305

**Comment by ViralBShah** [comment]
BTW, I just set up a Windows machine - so I'll check tomorrow if there is anything seriously wrong there as well.

**Pull request merged**